docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/gitlab.md

```diff
@@ -0,0 +1,111 @@
+---
+# Integrate TestCafe with GitLab
+
+This topic describes how to integrate TestCafe tests into a [GitLab](https://gitlab.com) project's build process.
+
+GitLab offers a built-in continuous integration service. You can read more about it in [GitLab documentation](https://docs.gitlab.com/ee/ci/quick_start/README.html).
+
+To run tests with TestCafe, you need a virtual machine with Node.js, TestCafe and browsers. Instead of installing them manually, you can use a [Docker](https://www.docker.com/) image.
+
+The easiest way to run TestCafe tests is [use an official TestCafe Docker image](#option-1---use-testcafe-docker-image) based on Alpine Linux with preinstalled Node.js, TestCafe, Firefox and Chromium.
+
+However, if you already have a Docker image prepared to deploy your web application and run tests, you can [install TestCafe to this image before testing](#option-2---install-testcafe-to-a-docker-image).
+
+## Option 1 - Use TestCafe Docker Image
+
+Open the `.gitlab-ci.yml` file that stores GitLab CI configuration. If you don't have this file yet, create it in the repository's root directory.
+
+Create a new [job](https://docs.gitlab.com/ee/ci/pipelines.html#jobs). To do this, add the following settings to `.gitlab-ci.yml` at the top level:
+
+```yaml
+e2e_tests:
+  image:
+    name: testcafe/testcafe
+    entrypoint: ["/bin/sh", "-c"]
+  script:
+    - /opt/testcafe/docker/testcafe-docker.sh firefox:headless,chromium tests/**/*
+```
+
+* `e2e_tests` - the job's name. You can choose any name you wish.
+* `image` - Docker image settings.
+* `name` - the TestCafe image's name. You can find it on [Docker Hub](https://hub.docker.com/r/testcafe/testcafe/).
+* `entrypoint` - overrides the image's [ENTRYPOINT](https://docs.docker.com/glossary/?term=ENTRYPOINT) to be `/bin/sh`. You need to specify this setting because the TestCafe image has the default `ENTRYPOINT` set to the `testcafe` command, while GitLab uses `sh` to run test scripts. The `-c` flag indicates that the subsequent argument is a command rather than a script.
+* `script` - the command that runs TestCafe tests. `/opt/testcafe/docker/testcafe-docker.sh` points to a script that prepares the environment to run a browser and starts TestCafe. Its arguments are standard TestCafe [command line parameters](../../using-testcafe/command-line-interface.md).
+
+For more information about the `.gitlab-ci.yml` syntax, see [Configuration of your jobs with .gitlab-ci.yml](https://docs.gitlab.com/ee/ci/yaml/README.html).
+
+Commit `.gitlab-ci.yml` and push the changes to the repository. Now you have a new [job](https://docs.gitlab.com/ee/ci/pipelines.html#jobs) that can be executed by any [GitLab Runner](https://docs.gitlab.com/ee/ci/runners/README.html).
+
+## Option 2 - Install TestCafe to a Docker Image
+
+If you use a Docker image that does not have TestCafe preinstalled, you can install it from `npm` before tests are run.
```

Comment:
You can install TestCafe from `npm` before tests are run if you use a Docker image that does not have TestCafe preinstalled.